### PR TITLE
Update Honeywell.cpp

### PR DESCRIPTION
**Pull request by deligoz**
I just did some overhaul, added cooling control and Fahrenheit support. need to update Honeywell.h as well. I don't know how to do a pull request for two files, so Honeywell.h will be a different pull request for this

### CODE REVIEW

**Comment by gizmocuz** [comment]
Thank you for your PR, But you can add the missing function in the header here as well
Just keep committing and pushing to this branch and it will be added here

**Comment by deligoz** [comment]
just figured out how to add the heather file to the pull request. please have a look

File: hardware/Honeywell.cpp [hunk 1]
```diff
@@ -240,48 +244,91 @@ void CHoneywell::GetThermostatData()
...
 
+			std::string units = device["units"].asString(); //:"Fahrenheit"
+
 			float temperature;
 			temperature = (float)device["indoorTemperature"].asFloat();
 			std::string desc = kRoomTempDesc;
 			stdreplace(desc, "[devicename]", deviceName);
-			SendTempSensor(10 * devNr + 1, 255, temperature, desc);
+			if(units == "Fahrenheit") {
+				SendTempSensor(10 * devNr + 1, 255, (temperature - 32) * 5/9, desc);
```
Comment (gizmocuz) [inline hardware/Honeywell.cpp]: Could you use the functions from Helper.h
ConvertToCelsius and if needed ConvertToFahrenheit

File: hardware/Honeywell.cpp [hunk 2]
```diff
@@ -123,19 +127,19 @@ void CHoneywell::Do_Work()
...
+	
 	const tRBUF *pCmd = reinterpret_cast<const tRBUF *>(pdata);
+	
 	if (pCmd->LIGHTING2.packettype == pTypeLighting2)
 	{
 		//Light command
-
-		int nodeID = pCmd->LIGHTING2.id3;
+		
+		int nodeID = pCmd->LIGHTING2.id4;
```
Comment (gizmocuz) [inline hardware/Honeywell.cpp:137]: Would this ID change cause all existing users to have new devices or cause other issues ?

File: hardware/Honeywell.cpp [hunk 3]
```diff
@@ -240,48 +244,91 @@ void CHoneywell::GetThermostatData()
 	}
 
 	int devNr = 0;
-	mDeviceList.clear();
+	mDeviceList.clear();
```
Comment (gizmocuz) [inline hardware/Honeywell.cpp]: Please undo whitespace at the end of the line

File: hardware/Honeywell.cpp [hunk 4]
```diff
@@ -314,7 +361,7 @@ void CHoneywell::SendSetPointSensor(const unsigned char Idx, const float Temp, c
 	thermos.id2 = 0;
 	thermos.id3 = 0;
 	thermos.id4 = Idx;
-	thermos.dunit = 0;
+	thermos.dunit = 0;
```
Comment (gizmocuz) [inline hardware/Honeywell.cpp]: Please undo whitespace at the end of the line

File: hardware/Honeywell.cpp [hunk 5]
```diff
@@ -324,23 +371,46 @@ void CHoneywell::SendSetPointSensor(const unsigned char Idx, const float Temp, c
...
+		if(bCommand){
+			nHeat = false;
+		}
+	}
+	
+	
 	std::string url = HONEYWELL_UPDATE_THERMOSTAT;
 	std::string deviceID = mDeviceList[idx]["deviceID"].asString();
-
+
```
Comment (gizmocuz) [inline hardware/Honeywell.cpp]: Please undo tab here

**Review by gizmocuz — COMMENTED** [review]
Thank you for the PR, just some small remarks

File: hardware/Honeywell.cpp (hunk 2 above)
Comment (deligoz) [reply hardware/Honeywell.cpp:137]: I actually change this id wrong at the previous commit. so this line is a roll back.

File: hardware/Honeywell.cpp (hunk 1 above)
Comment (deligoz) [reply hardware/Honeywell.cpp]: Thank you for pointing this out, just used them

File: hardware/Honeywell.cpp (hunk 3 above)
Comment (deligoz) [reply hardware/Honeywell.cpp]: done

File: hardware/Honeywell.cpp (hunk 4 above)
Comment (deligoz) [reply hardware/Honeywell.cpp]: done

File: hardware/Honeywell.cpp (hunk 5 above)
Comment (deligoz) [reply hardware/Honeywell.cpp]: done

**Comment by gizmocuz** [comment]
Thank you !!

**Pull request merged**